### ISSUE TITLE
Introducing concept of a 'disclosure' and disclosureDetails 

### ISSUE THREAD

**Issue by timgdavies** [openownership/data-standard#317]
# Summary of change proposal

Applications need to be able to judge when a given statement updates or replaces information they may already hold. 

The current `replacesStatement` model should be replaced with a `disclosureDetails` block and clear documentation on strategies that applications can adopt to manage changes in data, and data from different sources. 

This can also be considered as an alternative to replace the `isComponent` modelling introduced in BODS version 0.2, as it allows a set of statements to be logically grouped together. 

## Conceptual framework

*   A **publisher** is anyone producing beneficial ownership data.  
*   A **publisher database** reflects the ownership and control relationships a given publisher believes to be true. 
*   A **publisher ledger** contains all the statements made over time by a given publisher. 
*   An **applications ledger** contains all the statements, potentially from multiple sources, that an application has consumed. 
*   An **application database** contains the ownership and control relationships the application believes to be, or to have been, true. 

## Problem statement

BODS statements represent the information disclosed about ownership and control by a particular agent at a particular point in time. 

BODS encourages an audit ledger based approach, in which information is written out, but once written, is not updated.

Changes to assertions of ownership and control may be contained in later statements.

**Applications need to be able to judge when a given statement updates or replaces information they may already hold.**

## Worked example

_Note - not all the visual language is implemented at present in the diagrams below._

Person 1 **indirectly** owns 25% of Company B. This is disclosed via Source A at time t1: 

![image](https://user-images.githubusercontent.com/342624/83432067-3272ef00-a430-11ea-8300-16e56c985823.png)

([Source](https://gist.github.com/timgdavies/6b547245f17820f3105ca6de153f39f4#file-disclosure-0-json))

Person 1 takes a larger indirect share in Company B at time t2. This is also disclosed via Source A and time t3:

![image](https://user-images.githubusercontent.com/342624/83432078-39016680-a430-11ea-8b4e-a686d01c3ca0.png)

([Source](https://gist.github.com/timgdavies/6b547245f17820f3105ca6de153f39f4#file-disclosure-1-json))

At time t4, Company B has to disclose ownership and control through a different disclosure regime (Source B), requiring disclosure of a full ownership chain. Person 1 has dual nationality, and in this instance, provides a different nationality in their disclosure, as the form only provides space for one nationality:

![image](https://user-images.githubusercontent.com/342624/83432088-40287480-a430-11ea-9395-a7c1db387665.png)

([Source](https://gist.github.com/timgdavies/6b547245f17820f3105ca6de153f39f4#file-disclosure-2-json))

At time t5, Person 1 has to make their own disclosure, detailing all the companies they have a stake in. This is made via Source C:

![image](https://user-images.githubusercontent.com/342624/83432111-4a4a7300-a430-11ea-883e-f9cf7359eace.png)

([Source](https://gist.github.com/timgdavies/6b547245f17820f3105ca6de153f39f4#file-disclosure-3-json))

At time t6 a newspaper report reveals one of the other owners of Company B, with this data published by source D:

![image](https://user-images.githubusercontent.com/342624/83432129-51718100-a430-11ea-9183-524a68f8043e.png)

([Source](https://gist.github.com/timgdavies/6b547245f17820f3105ca6de153f39f4#file-disclosure-4-json))

Using human intelligence, we can assemble a reasonable picture of reality, by identifying that:

*   A 60% share specified later than a 25% ownership share must replace it
*   Person 2’s ownership of Company B is not incompatible with Person 1’s, and so can be added to it
*   Person 1’s disclosure of their ownership gives us extra information that is compatible with what Company B have disclosed, and which helps us understand the full ownership chain

This would give us the following ownership network:

![image](https://user-images.githubusercontent.com/342624/83432149-5b937f80-a430-11ea-9ece-0c81ae1f5eef.png)

([Source](https://gist.github.com/timgdavies/6b547245f17820f3105ca6de153f39f4#file-resolved-json))

We need a clear approach for how applications can also resolve multiple statements to get to a consistent **application database** state, given any collection (and order?) of incoming statements. 

## Prior approaches

The [alpha (0.0.1 - 0.0.3) versions of BODS](http://standard.openownership.org/en/0.0.1/schema.html) used statementGroups to collect sets of statements together, and allow one group of statements to replace a prior group of statements (`replacesStatementGroup`). 

This [was removed in 0.1](http://standard.openownership.org/en/0.1.0/schema/reference.html#replacesstatements), which removed structural statement grouping, but introduced `replacesStatement` to allow a statement to reference one or more prior statements that it replaced. 

Both these models require a publisher to keep track of past statements and the identifiers that have been assigned to them. This is potentially a significant demand to place on publishers. 

Version 0.2 introduced `isComponent` to all statements, and the ‘componentStatementIDs’ array to `OwnershipOrControlStatement`s as a means of logically grouping together statements that form part of a longer ownership chain. 

## Proposal: disclosureDetails

A **disclosure** is made up a collection of statements with a common **subject**. Each statement SHOULD have a `disclosureDetails` block within the statement metadata wrapper (see #136) that provides information on the subject of the disclosure, whether the disclosure is complete or partial, and whether the source believes the disclosure to be ‘active’ or ‘archived’. 

### In detail

The subject may be an entity - as in the case of a company required to disclosure it’s beneficial owners. 

The subject may be a person - as in the case of person declaring the entities they are beneficial owner of. 

A disclosure may be **complete** when it is asserted to include the full collection of statements about the subject required by a given disclosure regime.

A disclosure may be **partial** when it is _not_ asserted to include the full collection of statements about the subject required by a given disclosure regime. For example, when it provides only changed statements, or documents information gathered from secondary sources that may be incomplete. 

### Relating disclosures to common business processes

The concept of a disclosure maps to common business processes for beneficial ownership transparency.

*   A spreadsheet submitted by an extractives firm to their EITI national secretariat as part of EITI BO transparency constitutes an `entityComplete` disclosure. 

*   An annual confirmation statement submitted to the Persons of Significant Control register in the uk constitutes an `entityComplete` disclosure. 

*   Information published in a newspaper about the owners of a specific firm as a result of leaked information may constitute a `entityPartial` disclosure. 

*   An asset disclosure submitted by a politician at the start of their term of office, naming all the companies they control constitutes a `personComplete` disclosure.  

*   An update to an asset disclosure filed by the politician mid-way through their term of office, only listing new firms, may constitute a `personPartial` disclosure. 

It can help to think of a disclosure as all the statements extracted from a particular form submission. Generally forms ask for **all current information** or **updated information** about either a single **person** or a single **company**, with any information about intermediary entities or persons only required in so far as it relates to the primary subject of the disclosure. 

### How disclosureDetails works in practice

Every time a **publisher** receives new information, it should identify the **disclosure** that it belongs to. 

All the statements from a given **disclosure** should contain the same `disclosureDetails` block within the statement meta-data. 

```
"disclosureDetails": {
            "disclosureScope": "entityComplete",
            "subject": "243e8f5b-8699-4fac-9b88-21713e973951"
}
```

The `subject` of `disclosureDetails` is always a statement identifier (either an `EntityStatement` for `entityComplete` or `entityPartial` scopes, or a `PersonStatement` for `personComplete` or `personPartial` scopes). 

A consuming application should resolve the statement to an actual `entity` or `person` in its **application database**. For example (assuming #316 is implemented), statement `243e8f5b-8699-4fac-9b88-21713e973951` may have the following payload:

```
"entity":{
        "name": "Company B",
        "identifiers": [
            {
                "scheme": "UA-EDR",
                "id": "UA-XE-02"
            }
        ],
}
```

which an application can resolve to a record called "Company B" in it's database with the company identifier "UA-EDR-UA-XE-02" and the database primary key 'entity-123'. 

> Note: Real-world resolution of statements to identified entities or persons may need to be more sophisticated.

How an application choose to store this resolution is not specified, but, for example, it could for example update its **application ledger** copy of statements to include it as so:

```
"disclosureDetails": {
            "disclosureScope": "entityComplete",
            "subject": "243e8f5b-8699-4fac-9b88-21713e973951",
            "resolvesTo":"entity-123"
}
```

The application should then look for any past statements that have a `disclosureDetails` block that resolve to the same subject. 

If the disclosure scope is `entityComplete` **and** has a more recent `statementDate` then all past statements which resolve to "entity-123" should be marked as "archived", and the new statements marked as active. 

> ⚠️**CONSIDERATION:** Should statementDate be changed to disclosureDate and moved into the `disclosureDetails`? 

If the disclosure scope is `entityPartial` then the application has to apply some logic to decide whether to archive past statements or not. 

### ToDo

* [X] Expand on worked example JSON 
* [X] Write up how this can replace `isComponent` - done in issue #320

**Comment by timgdavies** [comment]
Note from https://github.com/openownership/openownership.beta/pull/39/files?file-filters%5B%5D=.md#r440237129 that the above might need to replace `personComplete` with `interestedPartyComplete` to handle cases where ownership chains do not terminate with a person, but instead terminate with a state owned enterprise or listed company.

**Comment by stevenday** [comment]
I'm not convinced that, across the entire ecosystem of data users, this is a simpler mechanism for marking updated/changed data over `replacesStatements`. It feels attractive for primary publishers of data (i.e. anyone who has data 'disclosed' to them directly) but it achieves that by hiding the complexity that consumers of more than one data source _will_ have to confront at some point. 

I think this will create a situation where every consumer app is forced to develop a change detection/reconciliation system and except in the simplest case - consuming data from a single, primary publisher - that system will be harder to build for the consumer than it would for the publisher. Because of that, my instinct is that it's more likely to result in naive systems unable to use/combine additional sources of data in the future. We've fallen into a similar trap ourselves on the OpenOwnership Register by treating OpenCorporates as a single source of truth about companies, rather than as one amongst many, and it's painful to get out of.

This kind of trade-off is something we should be conscious about making, as it might reflect an accurate balance of power/responsibility, putting the work where it can most feasibly be done. I think we should wait to see the outcomes of current publishers trying to collect and output BO data using BODS before deciding. 

My [own explorations](https://gist.github.com/stevenday/c1bd3e7eb1a195a7d760fcc6b4816911) of your scenarios suggest to me that `replacesStatements` is not harder to implement *if* your systems are designed around it from data collection onwards. It might be that `disclosures` are easier to retrofit onto an existing system, but I think if we're making such a retreat from complexity, we need to make sure we retreat far enough - this system still has implications on data collection that might be too onerous, if `replacesStatements` is.

(NB: I haven't considered the indirect/component situation yet, I'll comment on #320 separately)

**Comment by stevenday** [comment]
A further note that I've realised this suggestion introduces an additional ordering/relationship constraint to processing BODS data - you have to have seen (and processed) the subject of a disclosure before you can link other statements to it.

**Comment by timgdavies** [comment]
> This kind of trade-off is something we should be conscious about making, as it might reflect an accurate balance of power/responsibility, putting the work where it can most feasibly be done. I think we should wait to see the outcomes of current publishers trying to collect and output BO data using BODS before deciding.

I'm with you on the first part of this. 

I'm think though there is a strategic challenge with waiting for more real-world implementations on the current model, as the longer `replacesStatement` persists, the trickier change will be to other models in future (because standards quickly become brittle). 

> I think this will create a situation where every consumer app is forced to develop a change detection/reconciliation system and except in the simplest case - consuming data from a single, primary publisher - that system will be harder to build for the consumer than it would for the publisher.

The issue I see is that unless you can be confident that publishers all use `replacesStatement` correctly, an application ends up needing to implement a change detection system anyway. 

I'd be tempted to see this as a problem to be solved through investment in (reference implementation) intermediary open source software that can be used to process statements and/or plug into systems as a library to manage reconciliation - on the understanding that (a) change management is complex both for producers and consumers of data; (b) it may be more efficient to place costs onto intermediary than at either end of chain.

![image](https://user-images.githubusercontent.com/342624/88387213-f150e880-cda9-11ea-8008-64240787b1ef.png)


I also don't want to lose sight here of the communicative role of the shift to `disclosure` in trying to get from a focus on `statements` which have appeared over-abstract in communication about BODS, to the concept of making or discovering a disclosure, which is closer to everyday usage.

**Comment by timgdavies** [comment]
> A further note that I've realised this suggestion introduces an additional ordering/relationship constraint to processing BODS data - you have to have seen (and processed) the subject of a disclosure before you can link other statements to it.

True. Though this doesn't feel overly complex to implement? 

Noting that we also have implicit ordering constraint in `replacesStatement` in the same way, and if sticking with `replacesStatement` may need to write that up / include in validation checks.

**Comment by kd-ods** [comment]
> The issue I see is that unless you can be confident that publishers all use replacesStatement correctly, an application ends up needing to implement a change detection system anyway.

Indeed. But I think this gets to the heart of it. Better documentation and guidance will make the use of replacesStatements much, much easier. Plus supporting publishers to make their modelling choices and publishing schedules explicit in some kind of publication policy.

> I also don't want to lose sight here of the communicative role of the shift to disclosure in trying to get from a focus on statements which have appeared over-abstract in communication about BODS.

I think you have a point about the modelling of disclosures aligning - potentially helpfully - with people's real-world understanding. But I suspect the benefits will be short-lived once we start talking about entity- vs interestedParty-centric and full- vs partial- disclosures. These map very closely to business process and might prove unhelpful when caching things out for a data export. ReplacesStatements is more flexible and allows change over time to be represented at a finer resolution by the publisher (not pushing work onto the data user). The work pushed onto the user with the Disclosure model is a bit hidden:

> If the disclosure scope is entityPartial then the application has to apply some logic to decide whether to archive past statements or not.

Working out this logic could be far from trivial and again would depend on a solid implementation and a clear publication policy from the publisher.

Given where we're at, I'm with Steve's wait-and-see (and provide better documentation!) approach. Retrofitting disclosures onto the data model - and keeping Statements - seems like an untidy solution to a problem that we don't know exists.

**Comment by kathryn-ods** [comment]
@kd-ods can we close this ticket as we have implemented the alternative #317?

**Comment by kd-ods** [comment]
Yes, closing on the basis that this proposal has been overtaken / made redundant by the work to represent declarations (#482) and to represent changing information over time (#475) in BODS 0.4.
